### PR TITLE
Update Build scripts, Mariner image and latest drivers

### PR DESCRIPTION
**Pull request by jcamposeco**
Update Mariner image 
Update DOCA/OFED versions
Update latest SOC drivers
Update create bfb scripts

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,33 +10,33 @@ ADD rebuild_drivers /tmp
...
 RUN /usr/sbin/depmod -a $(/bin/ls -1 /lib/modules/ | head -1) || true
 
-RUN yum install -y ibacm ibutils2 infiniband-diags infiniband-diags-compat libibumad libibverbs-utils librdmacm librdmacm-utils libxpmem libxpmem-devel mft mft-oem mlnx-ethtool mlnx-fw-updater mlnx-iproute2 mlnx-libsnap mlx-regex mlxbf-bootctl mlxbf-bootimages mstflint ofed-scripts opensm opensm-devel opensm-libs opensm-static rdma-core rdma-core-devel srp_daemon ucx ucx-cma ucx-devel ucx-ib ucx-knem ucx-rdmacm xpmem mlnx-tools mlnx-dpdk mlnx-dpdk-devel dpcp libvma libvma-utils python3-grpcio python3-protobuf rxp-compiler
+RUN yum install -y ibacm ibutils2 infiniband-diags infiniband-diags-compat libibumad libibverbs libibverbs-utils librdmacm librdmacm-utils libxpmem libxpmem-devel mft mft-oem mlnx-ethtool mlnx-fw-updater mlnx-iproute2 mlnx-libsnap mlx-regex mlxbf-bootctl mlxbf-bootimages mstflint ofed-scripts opensm opensm-devel opensm-libs opensm-static perftest rdma-core rdma-core-devel srp_daemon ucx ucx-cma ucx-devel ucx-ib ucx-knem ucx-rdmacm xpmem mlnx-tools mlnx-dpdk mlnx-dpdk-devel dpcp libvma libvma-utils python3-grpcio python3-protobuf rxp-compiler
```
Comment (rlmenge) [inline Dockerfile:30]: Because libibverbs is [required for libibverbs-util](https://github.com/microsoft/CBL-Mariner/blob/9765a16d34a7f726960da6cb0d27841aa46c9366/SPECS/rdma-core/rdma-core.spec#L152), I don't believe this is necessary

File: Dockerfile [hunk 2]
```diff
@@ -10,33 +10,33 @@ ADD rebuild_drivers /tmp
...
 # The Mariner versions of these packages cause build failures. Use only the NVIDIA versions for now
-RUN yum install -y --disablerepo="mariner*" perftest libibverbs libreswan
+# RUN yum install -y --disablerepo="mariner*" perftest libibverbs libreswan
```
Comment (rlmenge) [inline Dockerfile:22]: I would remove if the Mariner packages are now sufficient :)

File: create_bfb [hunk 3]
```diff
@@ -159,7 +159,7 @@ GRUB_DEFAULT=saved
 GRUB_DISABLE_SUBMENU=true
 GRUB_TERMINAL=console
 GRUB_SERIAL_COMMAND="serial --unit=1"
-GRUB_CMDLINE_LINUX="crashkernel=auto  console=hvc0 console=ttyAMA0 earlycon=pl011,0x01000000 modprobe.blacklist=mlx5_core,mlx5_ib"
+GRUB_CMDLINE_LINUX="crashkernel=auto  console=hvc0 console=ttyAMA0 earlycon=pl011,0x01000000 modprobe.blacklist=mlx5_core,mlx5_ib net.ifnames=0 biosdevname=0"
```
Comment (rlmenge) [inline create_bfb:162]: Why was this needed out of curiosity?

File: create_bfb [hunk 4]
```diff
@@ -257,15 +257,15 @@ copy_rpm()
 }
 
 # Copy the content of RPM packages
-for package in  mlxbf-bfscripts util-linux dosfstools e2fsprogs kmod glibc coreutils pciutils usbutils chkconfig mft mft-oem kernel-mft tar which
+for package in  mlxbf-bfscripts util-linux dosfstools e2fsprogs kmod glibc coreutils pciutils usbutils chkconfig mft mft-oem kernel-mft tar which curl openssl $ADDON_RPMS
```
Comment (rlmenge) [inline create_bfb:260]: Where is this defined?

File: create_bfb (hunk 3 above)
Comment (jcamposeco) [reply create_bfb:162]: This change came from Vladimir

File: create_bfb (hunk 4 above)
Comment (jcamposeco) [reply create_bfb:260]: I think is added as an option to add new RPMs is not used as far as i know
Change is also coming from the Nvidia/Mariner repo

File: Dockerfile (hunk 1 above)
Comment (jcamposeco) [reply Dockerfile:30]: Will remove and verify

File: create_bfb (hunk 3 above)
Comment (jcamposeco) [reply create_bfb:162]: This is a link to the Mellanox/mariner repo https://github.com/Mellanox/bfb-mariner

File: Dockerfile (hunk 2 above)
Comment (jcamposeco) [reply Dockerfile]: removed

File: create_bfb (hunk 3 above)
Comment (jcamposeco) [reply create_bfb:162]: This is the feedback I got: "we use udev rules to set network interfaces names. Parameters above prevent from other utilities control over network interfaces names"
